share/eds/store.go

```diff
@@ -197,6 +201,10 @@ func (s *Store) Put(ctx context.Context, root share.DataHash, square *rsmt2d.Ext
 }
 
 func (s *Store) put(ctx context.Context, root share.DataHash, square *rsmt2d.ExtendedDataSquare) (err error) {
+	lk := &s.stripedLocks[root[len(root)-1]]
+	lk.Lock()
+	defer lk.Unlock()
+
```

Comment:
I wonder if we should put the lock in Put, so that we we don't track duplicates in traces and metrics

Reply:
Good point. Then `Has` check should be moved outside too.

Reply:
That's good, still not sure why would we even track them

Reply:
It is valuable from component centric approach, when you don't know who are callers of put and how many duplicates client are trying to store. Also Has check is not free, since it is badger lookup